### PR TITLE
Fix the Warnings in rst files when building Qlib's documentation

### PR DESCRIPTION
**Pull request by MaximSmolskiy**
Fix all warnings in `rst` files.

## Motivation and Context

Related to #1345.

## How Has This Been Tested?

- [x] Pass the test by running: `pytest qlib/tests/test_all_pipeline.py` under upper directory of `qlib`.
- [ ] If you are adding a new feature, test on your own test scripts.

## Screenshots of Test Results (if appropriate):
1. Pipeline test:
2. Your own tests:

## Types of changes

- [ ] Fix bugs
- [ ] Add new feature
- [x] Update documentation

### CODE REVIEW

**Comment by MaximSmolskiy** [comment]
@microsoft-github-policy-service agree

File: docs/advanced/task_management.rst [hunk 1]
```diff
@@ -18,7 +18,7 @@ With this module, users can run their ``task`` automatically at different period
 
 This whole process can be used in `Online Serving <../component/online.html>`_.
 
-An example of the entire process is shown `here <https://github.com/microsoft/qlib/tree/main/examples/model_rolling/task_manager_rolling.py>`_.
```
Comment (you-n-g) [inline docs/advanced/task_management.rst:21]: Why two underscores are necessary here? 

I didn't find  any reference in [rst docs](https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html)

Could you please share me any related references? 
Thanks

Comment (MaximSmolskiy) [reply docs/advanced/task_management.rst:21]: Previous version (with one underscore) raises `WARNING: Duplicate explicit target name: "...".`, I found the solution [here](https://github.com/sphinx-doc/sphinx/issues/3921#issuecomment-315581557)

Comment (you-n-g) [reply docs/advanced/task_management.rst:21]: @MaximSmolskiy
It really helpful to me. 
Thanks!

**Comment by you-n-g** [comment]
It still reports a lot of warnings in my environment.
image


But the current PR has already fixed a lot of issues is really helpful.
I can merge it first you want to fix it in other PRs in the future.

**Comment by MaximSmolskiy** [comment]
> It still reports a lot of warnings in my environment. image
> 
> But the current PR has already fixed a lot of issues is really helpful. I can merge it first you want to fix it in other PRs in the future.

Yes, in this PR I wanted to fix all warnings in `rst` files. Now there should be only `4` warnings with `rst` files:
`WARNING: document isn't included in any toctree`. I'm not sure if they need to be added, and if yes, that I can do it well (add them in index.rst in proper places with proper titles).

The the PR turned out to be not small (about hundred lines were changed).

In the future I will try to fix other errors/warnings in other PRs.

**Comment by you-n-g** [comment]
I think docs/developer/code_standard_and_dev_guide.rst should be included in the `index.rst`

Files in hidden (`docs/hidden/client.rst, docs/hidden/online.rst and docs/hidden/tuner.rst`) should be ignored.
Maybe [exclude_patterns](https://github.com/sphinx-doc/sphinx/issues/2467#issuecomment-211697586) will be useful

**Comment by you-n-g** [comment]
> I think docs/developer/code_standard_and_dev_guide.rst should be included in the `index.rst`
> 
> Files in hidden (`docs/hidden/client.rst, docs/hidden/online.rst and docs/hidden/tuner.rst`) should be ignored. Maybe [exclude_patterns](https://github.com/sphinx-doc/sphinx/issues/2467#issuecomment-211697586) will be useful

Would you like to update it in this PR? 
Otherwise I'll merge it first.

**Comment by MaximSmolskiy** [comment]
> > I think docs/developer/code_standard_and_dev_guide.rst should be included in the `index.rst`
> > Files in hidden (`docs/hidden/client.rst, docs/hidden/online.rst and docs/hidden/tuner.rst`) should be ignored. Maybe [exclude_patterns](https://github.com/sphinx-doc/sphinx/issues/2467#issuecomment-211697586) will be useful
> 
> Would you like to update it in this PR? Otherwise I'll merge it first.

I will try to do it in this PR soon

**Comment by MaximSmolskiy** [comment]
@you-n-g Now there are no warnings left with `rst` files and updated toc looks as follows
Снимок экрана от 2022-11-13 16-42-22

**Comment by you-n-g** [comment]
It looks great.
Do you think it will be more reasonable to put it before  `REFERENCE:`  and after `OTHER COMPONENTS/FEATURES/TOPICS:`?

File: docs/index.rst [hunk 2]
```diff
@@ -30,6 +30,11 @@ Document Structure
    Data Retrieval <start/getdata.rst>
    Custom Model Integration <start/integration.rst>
 
+.. toctree::
```
Comment (you-n-g) [inline docs/index.rst:33]: Do you think it will be more reasonable to put it before `REFERENCE`: and after `OTHER COMPONENTS/FEATURES/TOPICS:`?

Comment (MaximSmolskiy) [reply docs/index.rst]: Done

**Comment by you-n-g** [comment]
It looks great!
Thanks for your great efforts. 
Welcome to becoming one of [Qlib's contributors](https://github.com/microsoft/qlib/pulse)!

**Pull request merged**